blocklist-processing-engine/dockerfile

```diff
@@ -1,10 +1,45 @@
-FROM node:12
-WORKDIR /usr/src/app
-COPY . .
-RUN npm i
-# RUN mkdir build
-# RUN mkdir /build/data
-# COPY ./data/BlockedDataNumbers.txt ./build/data/BlockedDataNumbers.txt
-EXPOSE 10000
+FROM --platform=${TARGETPLATFORM:-linux/amd64} ghcr.io/openfaas/of-watchdog:0.8.4 as watchdog
+FROM node:12.16.1-alpine as builder
+
+ARG TARGETPLATFORM
+
+WORKDIR /opt/blocklist-rule-processing-engine
+
+ENV NPM_CONFIG_LOGLEVEL warn
+
+COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
+RUN chmod +x /usr/bin/fwatchdog
+
+RUN apk --no-cache add curl ca-certificates \
+    && addgroup -S app && adduser -S -g app app
+
+RUN apk add --no-cache -t build-dependencies git make gcc g++ python libtool autoconf automake \
+    && cd $(npm root -g)/npm \
+    && npm config set unsafe-perm true \
+    && npm install -g node-gyp
+
+COPY package.json package-lock.json* /opt/blocklist-rule-processing-engine/
+
+RUN npm ci
+
+RUN apk del build-dependencies
+
+COPY tsconfig.json /opt/blocklist-rule-processing-engine/
+COPY src /opt/blocklist-rule-processing-engine/src
+
 RUN npm run build
-CMD ["node", "build/index.js"]
\ No newline at end of file
+
+ENV cgi_headers="true"
```

Comment:
Got any idea what this means? Just curious

Reply:
yes this is config for the OpenFaas watchdog.